tests/common/from_params_test.py

```diff
@@ -862,3 +861,93 @@ def __init__(self, nested_class: Optional[Union[str, NestedClass]] = None):
             WrapperClass.from_params(
                 params=Params({"nested_class": {"wrong_varname": "varstring"}})
             )
+
+    def test_from_params_handles_base_class_kwargs(self):
+        class Foo(FromParams):
+            def __init__(self, a: int, b: str = None, **kwargs) -> None:
+                self.a = a
+                self.b = b
+                for key, value in kwargs.items():
+                    setattr(self, key, value)
+
+        foo = Foo.from_params(Params({"a": 2, "b": "hi"}))
+        assert foo.a == 2
+        assert foo.b == "hi"
+
+        foo = Foo.from_params(Params({"a": 2, "b": "hi", "c": {"2": "3"}}))
+        assert foo.a == 2
+        assert foo.b == "hi"
+        assert foo.c == {"2": "3"}
+
+    def test_from_params_base_class_kwargs_crashes_if_params_not_handled(self):
+        class Bar(FromParams):
+            def __init__(self, c: str = None) -> None:
+                self.c = c
+
+        class Foo(Bar):
+            def __init__(self, a: int, b: str = None, **kwargs) -> None:
+                super().__init__(**kwargs)
+                self.a = a
+                self.b = b
+                for key, value in kwargs.items():
+                    setattr(self, key, value)
```

Comment:
Do we need these two lines? I think this is redundant given that we're passing kwargs to `super().__init__()`.

Reply:
Yep, you're right, that was a copy-paste error.